src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/repository/LicenceSpecifications.kt

```diff
@@ -60,9 +80,19 @@ fun hasNomsIdIn(nomsIds: List<String>?): Specification<Licence>? = nomsIds?.let
 }
 
 fun hasResponsibleComIn(staffIds: List<Int>?): Specification<Licence>? = staffIds?.let {
-  return Specification<Licence> { root, _, builder ->
-    val joinOffenderManager: Join<Licence, CommunityOffenderManager> = root.join("responsibleCom", JoinType.INNER)
-    builder.and(joinOffenderManager.get<Int>("staffIdentifier").`in`(it))
+  return Specification<Licence> { root, query, criteriaBuilder ->
+    if (staffIds.isEmpty()) return@Specification null
+
+    query.distinct(true)
+
+    val predicates = LICENCE_KINDS_WITH_RESPONSIBLE_COM.map { licenceKind ->
```

Comment:
‏It builds an OR over several subclass joins. If those were INNER, the joins occur in the FROM clause and would require matches in all joined tables before the OR predicates are even evaluated. Using LEFT means non-matching joins return NULLs, so the row can still pass if any subclass predicate matches.